.github/workflows/publish.yml

```diff
@@ -22,76 +22,104 @@ concurrency:
           status: ${{ job.status }}
         env:
           SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+
       - uses: actions/checkout@v4
-      - uses: oleksiyrudenko/gha-git-credentials@v2-latest
+
+      - uses: oleksiyrudenko/gha-git-credentials@ac5b66bcf3873df4259c681d957d8c145980171f # v2.1.1
         with:
           token: '${{ secrets.GITHUB_TOKEN }}'
+
       - uses: actions/setup-node@v3
         with:
           node-version: '18.12.x'
+
       - name: Install dependencies
         run: npm ci
-      - name: Bump release version
-        run: npm version ${{ github.event.inputs.version-strategy }} -m "${{ github.event.inputs.title }}"
-      - name: Push main branch
-        run: git push origin main --follow-tags
-      - name: Set new release version
-        id: set-version
-        run: echo "version=v$(npm pkg get version | tr -d '\"')" >> "$GITHUB_OUTPUT"
-      - name: Set deployment SHA
-        id: set-deploy-sha
-        run: echo "deploy-sha=$(git rev-parse HEAD)" >> "$GITHUB_OUTPUT"
-      - name: Notify Slack about new package version
-        uses: ravsamhq/notify-slack-action@v2
+
+      - name: Get latest release
+        id: last-release
+        uses: pozetroninc/github-action-get-latest-release@d1dafdb6e338bdab109e6afce581a01858680dfb # v0.7.0
         with:
-          notification_title: "✅ constructor-ui-autocomplete package version bumped to ${{ steps.set-version.outputs.version }} successfully"
-          status: ${{ job.status }}
-        env:
-          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
+          repository: ${{ github.repository }}
+          excludes: "prerelease, draft"
+
+      - name: Create new git tag
+        uses: "./.github/actions/create-version-tag"
+        id: tag
+        with:
+          last-version: ${{ steps.last-release.outputs.release }}
+          version-strategy: ${{ inputs.version-strategy }}
+          slack-webhook-url: ${{ secrets.SLACK_WEBHOOK_URL }}
+
     outputs:
-        version: ${{ steps.set-version.outputs.version }}
-        deploy-sha: ${{ steps.set-deploy-sha.outputs.deploy-sha }}
+        version: ${{ steps.tag.outputs.next-version }}
+
   create_github_release:
     name: Create GitHub release
     needs: update_package_version
-    uses: ./.github/workflows/publish-create-github-release.yml
+    uses: ./.github/workflows/create-github-release.yml
     with:
       version: ${{ needs.update_package_version.outputs.version }}
       title: ${{ github.event.inputs.title }}
-      deploy-sha: ${{ needs.update_package_version.outputs.deploy-sha }}
     secrets: inherit
+    
   publish_to_npm:
     name: Publish to NPM
     needs: update_package_version
     uses: ./.github/workflows/publish-npm.yml
     with:
-      deploy-sha: ${{ needs.update_package_version.outputs.deploy-sha }}
+      version: ${{ needs.update_package_version.outputs.version }}
     secrets: inherit
+  
+  # disabled until we fully test it
+  publish_to_cdn:
```

Comment:
Would the workflow run this step if `publish_to_npm` fails?

Reply:
It would. The reason was to streamline the happy path, making it as fast as possible. But the sad path would increase its troubleshooting complexity. 

I'm not sure if the tradeoffs are worth it. We would have a longer publishing process, but I'm pretty sure that's not so time-sensitive. But I know that any surprise complexity trying to debug a workflow error will only make the sad path sadder.

Done b6f94df5ad01006b6570d6e71ef639658aa93b6d ✅ 

@pedro-lb any concerns about running all actions sequentially?

Reply:
No concerns at all. I think it's easier to manage if we run it sequentially.